### ISSUE TITLE
Compliance server in bjc fails connectivity tests and scans

### ISSUE THREAD

**Issue by SteveDelFante** [chef-cft/bjc#520]
"BJC Chef Demo (2.1.12)"

Connectivty test to Union for example"
```Failed to verify connectivity to sshKey://ubuntu@union.automate-demo.com:0 using login key 2de3086b-3e02-4f32-76dc-a4898770e0ea : exit status 1, 
WARN: Unresolved specs during Gem::Specification.reset:
      net-ssh (< 5.0, >= 2.6.5, >= 2.9)
      ffi (>= 1.0.1)
      multi_json (~> 1.10)
      rainbow (~> 2)
      diff-lcs (< 2.0, >= 1.2.0)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
/opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/backend.rb:44:in `rescue in create': Transport error, can't connect to 'ssh' backend: SSH session could not be established (RuntimeError)
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/backend.rb:16:in `create'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/runner.rb:64:in `configure_transport'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/runner.rb:56:in `initialize'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:238:in `new'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:238:in `run_command'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:169:in `detect'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/bin/inspec:12:in `<top (required)>'
	from /opt/chef-compliance/embedded/bin/inspec:22:in `load'
	from /opt/chef-compliance/embedded/bin/inspec:22:in `<main>'
Connection failed
```
Scan test from Union:

Unreachable Node...

```failed to verify connectivity to sshKey://ubuntu@union.automate-demo.com:0 using login key 2de3086b-3e02-4f32-76dc-a4898770e0ea : exit status 1, 
WARN: Unresolved specs during Gem::Specification.reset:
      net-ssh (< 5.0, >= 2.6.5, >= 2.9)
      ffi (>= 1.0.1)
      multi_json (~> 1.10)
      rainbow (~> 2)
      diff-lcs (< 2.0, >= 1.2.0)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
/opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/backend.rb:44:in `rescue in create': Transport error, can't connect to 'ssh' backend: SSH session could not be established (RuntimeError)
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/backend.rb:16:in `create'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/runner.rb:64:in `configure_transport'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/runner.rb:56:in `initialize'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:238:in `new'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:238:in `run_command'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/lib/inspec/cli.rb:169:in `detect'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /opt/chef-compliance/embedded/lib/ruby/gems/2.2.0/gems/inspec-1.16.1/bin/inspec:12:in `<top (required)>'
	from /opt/chef-compliance/embedded/bin/inspec:22:in `load'
	from /opt/chef-compliance/embedded/bin/inspec:22:in `<main>'```

**Comment by binamov** [comment]
workaround is to go to Compliance --> Settings --> Edit the listed id_rsa key

on the workstation from cmder, copy the contents of the id_rsa key into the clipboard by running `gc ~/.ssh/id_rsa | clip`

go back to the Compliance Edit the key UI and Paste what's in the clipboard. Hit Save. All should be good.

For a fix, we need to take a backup of the Compliance database with the new key and bake that into bjc.

**Comment by binamov** [comment]
fixed in 2.1.14
